### ISSUE TITLE
Remote play black screen when outside home wifi

### ISSUE THREAD

**Issue by edboug1** [unknownskl/xbox-xcloud-client#373]
Hi and thanks for this amazing program - it works perfectly for xcloud but sadly doesn't work for remote play. 

Remote play works on my iPhone but when accessing on my Mac only gets a black screen. I have tried restarting my xbox and reinstalling the Mac package but it doesn't make any difference.

Cheers.

**Comment by edboug1** [comment]
Update: 

I have just returned home and can connect from both my Mac and my iPhone when on my home network.

However, if I tether my Mac to my phone, I can only access from my phone again. I have tried updating my router as well as opening ports that have been suggested online, but sadly no success. 

Whilst trying to fix the issue, I came across [this post](https://github.com/unknownskl/xbox-xcloud-client/issues/342#issuecomment-1208328745), where it was suggested changing router settings to IPv6 to fix the issue. Very few ISPs in the UK actually support IPv6 so I am out of luck (as opening every other port didn't work).

As a fix in the meantime, (if it is ever possible for this app to work over IPv4), I have set up a local VPN which allows me to access my xbox using the Mac app.

Thank you again unknownskl.

**Comment by killam42** [comment]
Question how did you set up your VPN on your xbox?

**Comment by edboug1** [comment]
I have a raspberry pi I wasn't using and have set that up as a VPN - so I'm essentially connecting from my laptop, to my raspberry pi inside my home network and that's connecting to my xbox, which gets around any port/IPv6 issues.

**Comment by killam42** [comment]
Oh ok then

**Comment by trashbus99** [comment]
Did you try connecting without the VPN?  I'm pretty sure when you login to Microsoft,  Microsoft tells the client (your laptop) the ip address that Microsoft sees the xbox reporting to its network and not the vpn tunnel.

**Comment by edboug1** [comment]
Yeah I tried tethering to my phone from my Mac when at home, without the VPN, and it didn't work. When I repeated the same thing but with the VPN turned on, it worked perfectly. For clarity, my laptop is connected to my VPN inside my home network (and by tethering this is mimicking being outside my home network), whereas my xbox is connected as normal.

I believe the issue is related to my "xbox stream" leaving my home network when not encapsulated by Teredo (as the official app does, which is described at [this link](https://github.com/unknownskl/xbox-xcloud-client/issues/342#issuecomment-1208328745)). By encapsulating the "xbox stream" within the VPN, this gets around any IPv6 router issues. I would imagine that if this app used Teredo in a similar way to the official app then it would work, but I don't know if that is in the developer's plans.

Even if what you said is true, given the data flow is as below, once my laptop's network requests leave the VPN, they are inside my home network and can immediately access the IP address that Microsoft has given my laptop.

Data flow:
Laptop receives xbox IP address from MS (outside) -> Laptop sends request via VPN (outside) -> VPN passes on request (inside) -> xbox processes request and sends response (inside) -> VPN passes on response (inside) -> laptop receives response (outside)
